Update for parity with changes in record functions

Plugin does not build with coredns main branch, I tried at coredns commit a929b0b1ecbb0f87b0bd0cd0a40ce38893dce9c0

```
lauralorenz@lauralorenz:~/coredns/coredns$ make
CGO_ENABLED=0  go build -v -ldflags="-s -w -X github.com/coredns/coredns/coremain.GitCommit=a929b0b1-dirty" -o coredns
github.com/coredns/multicluster
# github.com/coredns/multicluster
../../go/pkg/mod/github.com/coredns/multicluster@v0.0.0-20211130154347-488880e11fc0/multicluster.go:131:18: assignment mismatch: 2 variables but plugin.A returns 3 values
../../go/pkg/mod/github.com/coredns/multicluster@v0.0.0-20211130154347-488880e11fc0/multicluster.go:133:18: assignment mismatch: 2 variables but plugin.AAAA returns 3 values
../../go/pkg/mod/github.com/coredns/multicluster@v0.0.0-20211130154347-488880e11fc0/multicluster.go:135:18: assignment mismatch: 2 variables but plugin.TXT returns 3 values
../../go/pkg/mod/github.com/coredns/multicluster@v0.0.0-20211130154347-488880e11fc0/multicluster.go:148:12: assignment mismatch: 2 variables but plugin.A returns 3 values
make: *** [Makefile:16: coredns] Error 2
```

Appears to be due to function signature changes for records in https://github.com/coredns/coredns/pull/4715 meaning this plugin is currently incompatible as of coredns 1.9.1.

--- a/multicluster.go
+++ b/multicluster.go
@@ -136,10 +136,10 @@ func (m MultiCluster) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 	case dns.TypeTXT:
 		records, truncated, err = plugin.TXT(ctx, &m, zone, state, nil, plugin.Options{})
 	case dns.TypeSRV:
-		records, truncated, extra, err = plugin.SRV(ctx, &m, zone, state, plugin.Options{})
+		records, extra, err = plugin.SRV(ctx, &m, zone, state, plugin.Options{})
 	case dns.TypeNS:
 		if state.Name() == zone {
-			records, truncated, extra, err = plugin.NS(ctx, &m, zone, state, plugin.Options{})
+			records, extra, err = plugin.NS(ctx, &m, zone, state, plugin.Options{})
 			break
 		}
 		fallthrough

--- a/multicluster.go
+++ b/multicluster.go
@@ -4,6 +4,9 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"strings"
+	"time"
+
 	"github.com/coredns/coredns/plugin/etcd/msg"
 	"github.com/coredns/coredns/plugin/pkg/dnsutil"
 	"github.com/coredns/coredns/plugin/pkg/fall"
@@ -15,8 +18,6 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	apiv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
 	"sigs.k8s.io/mcs-api/pkg/client/clientset/versioned/typed/apis/v1alpha1"
-	"strings"
-	"time"
 
 	"github.com/coredns/coredns/plugin"
 	clog "github.com/coredns/coredns/plugin/pkg/log"
@@ -121,23 +122,24 @@ func (m MultiCluster) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 	state.Zone = zone
 
 	var (
-		records []dns.RR
-		extra   []dns.RR
-		err     error
+		records   []dns.RR
+		extra     []dns.RR
+		truncated bool
+		err       error
 	)
 
 	switch state.QType() {
 	case dns.TypeA:
-		records, err = plugin.A(ctx, &m, zone, state, nil, plugin.Options{})
+		records, truncated, err = plugin.A(ctx, &m, zone, state, nil, plugin.Options{})
 	case dns.TypeAAAA:
-		records, err = plugin.AAAA(ctx, &m, zone, state, nil, plugin.Options{})
+		records, truncated, err = plugin.AAAA(ctx, &m, zone, state, nil, plugin.Options{})
 	case dns.TypeTXT:
-		records, err = plugin.TXT(ctx, &m, zone, state, nil, plugin.Options{})
+		records, truncated, err = plugin.TXT(ctx, &m, zone, state, nil, plugin.Options{})
 	case dns.TypeSRV:
-		records, extra, err = plugin.SRV(ctx, &m, zone, state, plugin.Options{})
+		records, truncated, extra, err = plugin.SRV(ctx, &m, zone, state, plugin.Options{})
 	case dns.TypeNS:
 		if state.Name() == zone {
-			records, extra, err = plugin.NS(ctx, &m, zone, state, plugin.Options{})
+			records, truncated, extra, err = plugin.NS(ctx, &m, zone, state, plugin.Options{})
 			break
 		}
 		fallthrough
@@ -145,7 +147,7 @@ func (m MultiCluster) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 		// Do a fake A lookup, so we can distinguish between NODATA and NXDOMAIN
 		fake := state.NewWithQuestion(state.QName(), dns.TypeA)
 		fake.Zone = state.Zone
-		_, err = plugin.A(ctx, &m, zone, fake, nil, plugin.Options{})
+		_, _, err = plugin.A(ctx, &m, zone, fake, nil, plugin.Options{})
 	}
 
 	if m.IsNameError(err) {
@@ -168,6 +170,7 @@ func (m MultiCluster) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 
 	message := new(dns.Msg)
 	message.SetReply(r)
+	message.Truncated = truncated
 	message.Authoritative = true
 	message.Answer = append(message.Answer, records...)
 	message.Extra = append(message.Extra, extra...)

